cli/cmd/wallet.go

```diff
@@ -0,0 +1,254 @@
+			var passphrase string
+			err := survey.AskOne(common.PromptPassphrase, &passphrase)
+			cobra.CheckErr(err)
+
+			walletCfg := &config.Wallet{
+				Kind: walletKind,
+			}
+			err = walletCfg.SetConfigFromFlags()
+			cobra.CheckErr(err)
+
+			err = cfg.Wallets.Create(name, string(passphrase), walletCfg)
+			cobra.CheckErr(err)
+
+			err = cfg.Save()
+			cobra.CheckErr(err)
+		},
+	}
+
+	walletShowCmd = &cobra.Command{
+		Use:   "show <name>",
+		Short: "Show public wallet information",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			name := args[0]
+
+			wallet := common.LoadWallet(config.Global(), name)
+			showPublicWalletInfo(wallet)
+		},
+	}
+
+	walletRmCmd = &cobra.Command{
+		Use:     "rm <name>",
+		Aliases: []string{"remove"},
+		Short:   "Remove an existing wallet",
+		Args:    cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name := args[0]
+
+			// Early check for whether the wallet exists so that we don't ask for confirmation first.
+			if _, exists := cfg.Wallets.All[name]; !exists {
+				cobra.CheckErr(fmt.Errorf("wallet '%s' does not exist", name))
+			}
+
+			fmt.Printf("WARNING: Removing the wallet will ERASE secret key material!\n")
+			fmt.Printf("WARNING: THIS ACTION IS IRREVERSIBLE!\n")
+
+			var result string
+			confirmText := fmt.Sprintf("I really want to remove wallet %s", name)
+			prompt := &survey.Input{
+				Message: fmt.Sprintf("Enter '%s' (without quotes) to confirm removal:", confirmText),
+			}
+			survey.AskOne(prompt, &result)
+
+			if result != confirmText {
+				cobra.CheckErr("Aborted.")
+			}
+
+			err := cfg.Wallets.Remove(name)
+			cobra.CheckErr(err)
+
+			err = cfg.Save()
+			cobra.CheckErr(err)
+		},
+	}
+
+	walletSetDefaultCmd = &cobra.Command{
+		Use:   "set-default <name>",
+		Short: "Sets the given wallet as the default wallet",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name := args[0]
+
+			err := cfg.Wallets.SetDefault(name)
+			cobra.CheckErr(err)
+
+			err = cfg.Save()
+			cobra.CheckErr(err)
+		},
+	}
+
+	walletImportCmd = &cobra.Command{
+		Use:   "import <name>",
+		Short: "Import an existing wallet",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := config.Global()
+			name := args[0]
+
+			if _, exists := cfg.Wallets.All[name]; exists {
+				cobra.CheckErr(fmt.Errorf("wallet '%s' already exists", name))
+			}
+
+			// Ask for import details.
+			questions := []*survey.Question{
+				{
+					Name: "kind",
+					Prompt: &survey.Select{
+						Message: "Import kind:",
+						Options: wallet.ImportKinds(),
+					},
+				},
+				{
+					Name: "algorithm",
+					Prompt: &survey.Select{
+						Message: "Algorithm:",
+						Options: []string{"ed25519"},
+						Default: "ed25519",
+					},
+				},
+				{
+					Name:   "data",
+					Prompt: &survey.Multiline{Message: "Data to import:"},
```

Comment:
here's a part that we could make friendlier imo. this question depends on what what you chose for 'kind,' but we cram it into a universal 'data' question. if we already knew, we could give more detailed instructions in the prompt, e.g. "mnemonic" for mnemonic imports, we could additionally ask for account number, or we could say that the private key must be base64

maybe we could split this off into a separate `Ask`

Reply:
nice, thanks